### PR TITLE
Add example how to register commands via DI

### PR DESCRIPTION
**Pull request by DanielSiepmann**
Relates: #845

### CODE REVIEW

**Comment by DanielSiepmann** [comment]
- [ ] Cherry Pick 10.4

**Comment by DanielSiepmann** [comment]
Thanks @brotkrueml for reminding me.

I've reworked the page and mentioned all the deprecations / features with proper syntax.
Also mentioned deprecation inline.

File: Documentation/ApiOverview/CommandControllers/Index.rst [hunk 1]
```diff
@@ -10,32 +9,57 @@
...
 ==============================
 
-It is possible to run some TYPO3 CMS scripts from the
-command line. This makes it possible - for example -
-to set up cronjobs.
-
-TYPO3 uses Symfony commands to provide an easy to use, well-documented API for
-writing CLI (command line interface) commands.
+It is possible to run TYPO3 CMS scripts from the command line.
+This makes it possible to set up cronjobs.
```
Comment (sypets) [inline Documentation/ApiOverview/CommandControllers/Index.rst:13]: Minor nitpick: "possible" twice. I will fix.

File: Documentation/ApiOverview/CommandControllers/Index.rst [hunk 2]
```diff
@@ -51,19 +75,17 @@ Creating a new Symfony Command in Your Extension
 
    :php:`configure()`
       As the name would suggest allows to configure the command.
-      Allows to add a description or a help text, or mandatory and optional
-      arguments and parameters defined.
+      Allows to add a description or a help text,
+      or mandatory and optional arguments and parameters defined.
```
Comment (sypets) [inline Documentation/ApiOverview/CommandControllers/Index.rst:79]: minor nitpick (this was already this way before): how can arguments be mandatory _and_ optional? I would just simplify it (the point that you can have mandatory or optional arguments will reveal itself in the code snippet below).

> Allows to add a description, a help text or define arguments for the command.

File: Documentation/ApiOverview/CommandControllers/Index.rst [hunk 3]
```diff
@@ -151,16 +194,15 @@ this by setting `schedulable` to `false` in :file:`Configuration/Commands.php`::
...
    ];
 
-
 Initialize Backend User
 -----------------------
 
-If anything related to :ref:`DataHandler  <datahandler-basics>` and backend
-permission handling is necessary, you should call this initialization
-method once in your :php:`execute()` function::
+Backend user can be initialized with this call inside :php:`execute()` method::
```
Comment (sypets) [inline Documentation/ApiOverview/CommandControllers/Index.rst:200]: -> 

> A backend user ...

File: Documentation/ApiOverview/CommandControllers/Index.rst [hunk 4]
```diff
@@ -173,30 +215,11 @@ The above example can be run via command line:
...
 
-.. tip::
-
-   If you installed TYPO3 without Composer, the path for the executable
-   is :file:`typo3/sysext/core/bin/typo3`.
-
-
-.. _symfony-console-commands-scheduler:
-
-Running the Command From the Scheduler
```
Comment (sypets) [inline Documentation/ApiOverview/CommandControllers/Index.rst:191]: Why did you remove this completely?

Comment (DanielSiepmann) [reply Documentation/ApiOverview/CommandControllers/Index.rst:191]: Because this is already mentioned above on line 168:
> By default, the command can be used in the scheduler too.
